src/httpd_jsonapi.c

```diff
@@ -3339,6 +3341,20 @@ jsonapi_reply_library_tracks_put_byid(struct httpd_request *hreq)
         return HTTP_INTERNAL;
     }
 
+  // retreive via "/api/search?type=tracks&expression=flag+=+1"
+  param = evhttp_find_header(hreq->query, "flag");
+  if (param)
+    {
+      ret = safe_atoi32(param, &val);
+      if (ret < 0 || (ret == 0 && (val < FLAG_KIND_NA || val >= FLAG_KIND_MAX)) )
```

Comment:
I don't know why this function uses `int val`, since rating also cannot be non-zero and the db_ functions take uint32_t as argument. I suggest you change val to uint32_t, use safe_atou32() also for rating above and then drop the range check, except for `<= DB_FILES_RATING_MAX` above.

Reply:
On second thought, the value stored in the db is probably int32, so maybe a better fix is to change the db_ functions to take int32 arguments and then stick with int32 here (incl. negative value range checks).

Reply:
its a `INTEGER` in db definition but represented as a`uint32_t` (like `rating`) - allowing < 0 would not less us do bitwise enumeration of the usermark types.

i'll make the change so its like `rating` handling above but i dislike the _silent failure_ (ie range check fails but response is success and not HTTP_BADREQUEST0